src/services/services.ts

```diff
@@ -2807,6 +2810,22 @@ namespace ts {
                     }
                 }
 
+                // Special case for function expression and class expression because despite sometimes having a name, the binder
+                // binds them to a symbol with the name "__function" and "__class" respectively. However, for completion entry, we want
+                // to display its declared name rather than "__function" and "__class".
+                //      var x = function foo () {
+                //          fo$  <- completion list should contain local name "foo"
+                //      }
+                //      foo$ <- completion list should not contain "foo"
+                if (displayName === "__function" || displayName === "__class") {

```

Comment:
Look into `getDeclaredName` for logic, including the above for the `default` export